libbeat/processors/add_process_metadata/add_process_metadata_test.go

```diff
@@ -1102,3 +1106,220 @@ func TestV2CID(t *testing.T) {
+				},
+			},
+			4: {
+				V2: map[string]cgroup.ControllerPath{
+					"Podman": {IsV2: true, ControllerPath: "/user.slice/user-1000.slice/user@1000.service/user.slice/libpod-conmon-ee059a097566fdc5ac9141bfcdfbed0c972163da891de076e0849d7b53597aac.scope"},
+				},
+			},
+			5: {
+				V2: map[string]cgroup.ControllerPath{
+					"Docker": {IsV2: true, ControllerPath: "/docker/485776c9f6f2c22e2b44a2239b65471d6a02701b54d1cb5e1c55a09108a1b5b9"},
+				},
+			},
+		}
+		return testMap[pid], nil
+	}
+
+	for _, test := range []struct {
+		description             string
+		config, event, expected mapstr.M
+		err, initErr            error
+	}{
+		{
+			description: "container.id only",
+			config: mapstr.M{
+				"match_pids":     []string{"system.process.ppid"},
+				"include_fields": []string{"container.id"},
+			},
+			event: mapstr.M{
+				"system": mapstr.M{
+					"process": mapstr.M{
+						"ppid": "1",
+					},
+				},
+			},
+			expected: mapstr.M{
+				"system": mapstr.M{
+					"process": mapstr.M{
+						"ppid": "1",
+					},
+				},
+				"container": mapstr.M{
+					"id": "80d85a3a585f1575028ebe468d83093c301eda20d37d1671ff2a0be50fc0e460",
+				},
+			},
+		},
+		{
+			description: "container.id only",
+			config: mapstr.M{
+				"match_pids":     []string{"system.process.ppid"},
+				"include_fields": []string{"container.id"},
+			},
+			event: mapstr.M{
+				"system": mapstr.M{
+					"process": mapstr.M{
+						"ppid": "2",
+					},
+				},
+			},
+			expected: mapstr.M{
+				"system": mapstr.M{
+					"process": mapstr.M{
+						"ppid": "2",
+					},
+				},
+				"container": mapstr.M{
+					"id": "e01a26336924e2fb8089bcf4cf943954fd9ea616cc5678f38f65928307979459",
+				},
+			},
+		},
+		{
+			description: "container.id only",
+			config: mapstr.M{
+				"match_pids":     []string{"system.process.ppid"},
+				"include_fields": []string{"container.id"},
+			},
+			event: mapstr.M{
+				"system": mapstr.M{
+					"process": mapstr.M{
+						"ppid": "3",
+					},
+				},
+			},
+			expected: mapstr.M{
+				"system": mapstr.M{
+					"process": mapstr.M{
+						"ppid": "3",
+					},
+				},
+				"container": mapstr.M{
+					"id": "80d85a3a585f1575028ebe468d83093c301eda20d37d1671ff2a0be50fc0e460",
+				},
+			},
+		},
+		{
+			description: "container.id only",
+			config: mapstr.M{
+				"match_pids":     []string{"system.process.ppid"},
+				"include_fields": []string{"container.id"},
+			},
+			event: mapstr.M{
+				"system": mapstr.M{
+					"process": mapstr.M{
+						"ppid": "4",
+					},
+				},
+			},
+			expected: mapstr.M{
+				"system": mapstr.M{
+					"process": mapstr.M{
+						"ppid": "4",
+					},
+				},
+				"container": mapstr.M{
+					"id": "ee059a097566fdc5ac9141bfcdfbed0c972163da891de076e0849d7b53597aac",
+				},
+			},
+		},
+		{
+			description: "container.id only",
+			config: mapstr.M{
```

Comment:
How about a test case where the user has configured a custom `cgroup_prefixes`. I think we might be causing a breaking change due to how the configured `cgroup_regex` takes precedence.

Reply:
I think this is a breaking change for `cgroup_prefixes` usage. This config will never be considered as the priority is always on `cgroup_regex` and it is checked first when enriching `cid`.
https://github.com/elastic/beats/blob/main/libbeat/processors/add_process_metadata/gosigar_cid_provider.go#L116

Reply:
Added the test